x-pack/plugins/apm/server/routes/settings/custom_link.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import * as t from 'io-ts';
+import {
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  TRANSACTION_NAME,
+  TRANSACTION_TYPE
+} from '../../../common/elasticsearch_fieldnames';
+import { createRoute } from '../create_route';
+import { setupRequest } from '../../lib/helpers/setup_request';
+import { createOrUpdateCustomLink } from '../../lib/settings/custom_link/create_or_update_custom_link';
+import { deleteCustomLink } from '../../lib/settings/custom_link/delete_custom_link';
+import { listCustomLinks } from '../../lib/settings/custom_link/list_custom_links';
+
+const FilterOptions = t.partial({
```

Comment:
We normally suffix the runtime types with `Rt`:
```suggestion
const FilterOptionsRt = t.partial({
```

Then you can rename `FilterOptionsType` to `FilterOptions`